flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStreamWithResultProvider.java

```diff
@@ -172,4 +167,30 @@ public CheckpointStreamWithResultProvider create(
 			return new CheckpointStreamWithResultProvider.PrimaryStreamOnly(primaryOut);
 		}
 	}
+
+	/**
+	 * Helper method that takes a {@link SnapshotResult<StreamStateHandle>} and a {@link KeyGroupRangeOffsets} and
+	 * creates a {@link SnapshotResult<KeyGroupsStateHandle>} by combining the key groups offsets with all the
+	 * present stream state handles.
+	 */
+	static SnapshotResult<KeyedStateHandle> toKeyedStateHandleSnapshotResult(
+		@Nullable SnapshotResult<StreamStateHandle> snapshotResult,
```

Comment:
I think this parameter should not be `Nullable`. If it is null, then we don't have to call this method.

Reply:
Already changed in my updated branch.